test/e2e/upgrade/service/service.go

```diff
@@ -261,28 +261,28 @@ func locateService(svc *v1.Service) string {
 
 // TestReachableHTTPWithMinSuccessCount tests that the given host serves HTTP on the given port for a minimum of successCount number of
 // counts at a given interval. If the service reachability fails, the counter gets reset
-func TestReachableHTTPWithMinSuccessCount(host string, port int, successCount int, interval time.Duration, timeout time.Duration) {
+func TestReachableHTTPWithMinSuccessCount(host string, port int, successCount int, timeout time.Duration) {
+	consecutiveSuccessCnt := 0
 	pollfn := func() (bool, error) {
 		result := e2enetwork.PokeHTTP(host, port, "/echo?msg=hello",
 			&e2enetwork.HTTPPokeParams{
 				BodyContains:   "hello",
 				RetriableCodes: []int{},
 			})
 		if result.Status == e2enetwork.HTTPSuccess {
-			return true, nil
+			consecutiveSuccessCnt++
+			return consecutiveSuccessCnt >= successCount, nil
+		} else {
+			consecutiveSuccessCnt = 0
 		}
 		return false, nil // caller can retry
 	}
 
-	for retryCount := 0; retryCount < successCount; retryCount++ {
-		if err := wait.PollImmediate(framework.Poll, timeout, pollfn); err != nil {
-			if err == wait.ErrWaitTimeout {
-				framework.Logf("Could not reach HTTP service through %v:%v after %v", host, port, timeout)
-			} else {
-				framework.Logf("Failed to reach HTTP service through %v:%v: %v", host, port, err)
-			}
-			retryCount = 0
+	if err := wait.PollImmediate(framework.Poll, timeout, pollfn); err != nil {
+		if err == wait.ErrWaitTimeout {
```

Comment:
You need to error if poll immediate fails.  That's why my previous comment inlined the poll function (which is weird anyway).

Reply:
I thought `framework.Failf` will cause panic and the test to fail?